### PR TITLE
Upper case hex for address display/debug

### PR DESCRIPTION
**Pull request by tnowacki**
- Changed display of address to upper hex

## Motivation

- We canonically use upper case hex in Move programs
- I will drop this PR if we use display/debug in anywhere critical

## Test Plan

- updated source lang/cli tests

### CODE REVIEW

**Comment by sblackshear** [comment]
I like this *much* better aesthetically but am unsure if it will have downstream effects on the CLI/json-rpc/elsewhere. CC @xli @kchalkias @davidiw to see if there are any anticipated bad effects

**Comment by tnowacki** [comment]
The spots for access paths seem to call out to lower hex directly, via `{:#x}`, so we should be in the clear there? 

Definitely ready to drop this quickly if it turns out to be too entrenched.

File: language/move-lang/src/shared/mod.rs [hunk 1]
```diff
@@ -72,19 +72,19 @@ impl AsRef<[u8]> for Address {
...
 
 impl fmt::Debug for Address {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "0x{:#x}", self)
+        write!(f, "0x{:#X}", self)
     }
 }
 
-impl fmt::LowerHex for Address {
+impl fmt::UpperHex for Address {
```
Comment (sblackshear) [inline language/move-lang/src/shared/mod.rs:85]: Does implementing `UpperHex` give you `LowerHex` for free? And if not, can we keep the `LowerHex` impl?

**Comment by sblackshear** [comment]
> The spots for access paths seem to call out to lower hex directly, via `{:#x}`, so we should be in the clear there?

I really hope the hashing for access paths uses the raw byes of the address instead of a stringified version...

File: language/move-lang/src/shared/mod.rs (hunk 1 above)
Comment (tnowacki) [reply language/move-lang/src/shared/mod.rs:85]: I'll keep both, yeah

**Comment by tnowacki** [comment]
> > The spots for access paths seem to call out to lower hex directly, via `{:#x}`, so we should be in the clear there?
> 
> I really hope the hashing for access paths uses the raw byes of the address instead of a stringified version...

It probably does? No idea, its just the one spot I right away saw a build error when I removed the lower hex trait impl

**Comment by xli** [comment]
I believe it is only used in test, as long as all tests pass, should be fine, although probably will have some client sdk tests broken as comparing with lower cases.

**Comment by tnowacki** [comment]
Found a bunch of weird stuff in the faucet tests where they used string representations of account addresses. After fixing that it looks like everything is running

**Review by sblackshear — APPROVED** [review]
The stronger typing in the faucet looks like an improvement to me.

**Comment by dmitri-perelman** [comment]
As an anecdote, AOS had some (unexpected and non-planned) dependency on address being lower-case... (the HSMs would typically use the addresses written in the approval policies that come with addresses written in in lower case, but during the initialization we'd use `AccountAddress::to_string()`, which changed to be upper case after the display changed). 
Obviously, an AOS bug, we should have not depended on that, but it was fun to investigate :)

**Comment by metajack** [comment]
This broke our Rosetta implementation as it didn't change the output everywhere it appears.

I'll note that Coinbase considers addresses case sensitive due to [Ethereum's checksumming](https://coincodex.com/article/2078/ethereum-address-checksum-explained/), which is some support for us moving to all lowercase as canonical in case we ever want to do similar checksumming.

**Pull request closed without merging**